tests/rptest/remote_scripts/stream_verifier_txn.py

```diff
@@ -0,0 +1,974 @@
+                'transactional.id': t.transaction_id
+            })
+            t.producer.init_transactions()
+            self.topics[t.name] = t
+
+    def _send_transaction(self, logger: logging.Logger,
+                          topic: TopicStatus) -> int:
+        """
+            Transactional message sent. Uses simple indexed generator
+        """
+        sent_count = 0
+
+        def acked(err: ck.KafkaError, msg: ck.Message):
+            """
+                Unsafe callback that fills up delivery reports
+                Can be modified to send reports externally
+
+                err: error class from Kafka
+                msg: message that was sent
+            """
+            t = msg.topic()
+            pt = msg.partition()
+            pt = pt if pt else 'N'
+            k = msg.key().decode()
+            v = msg.value().decode()
+
+            if err is not None:
+                # log delivery error locally
+                logger.error(f"[{t}({pt}):{k}/{v}] {err.str()}")
+                # update delivery report
+                self.delivery_reports[f"{t}-{k}"] = {
+                    "latency": msg.latency(),
+                    "outcome": err.str()
+                }
+
+            else:
+                # Saving all delivery reports turned off for now
+                # self.delivery_reports[f"{t}-{k}"] = {
+                #     "partition": pt,
+                #     "latency": msg.latency(),
+                #     "offset": msg.offset(),
+                #     "outcome": "OK"
+                # }
+                pass
+            return
+
+        def time_since_last_msg() -> int:
+            diff_ms = datetime.now().timestamp() - topic.last_message_ts
+            return int(diff_ms * 1000)
+
+        msg_gen = MessageGenerator()
+        for key, value in msg_gen.gen_indexed_messages(
+                topic.index, topic.msgs_per_transaction):
+            # Handle message rate
+            if topic.msgs_rate_ms > 0:
+                _time_since = time_since_last_msg()
+                if _time_since < topic.msgs_rate_ms:
+                    wait_time = (topic.msgs_rate_ms - _time_since) / 1000
+                    logger.debug(f"...waiting {wait_time}s "
+                                 "before sending message")
+                    sleep(wait_time)
+
+            # Async message sending
+            topic.producer.begin_transaction()
+            try:
+                topic.producer.produce(topic.name,
+                                       key=key,
+                                       value=value,
+                                       callback=acked)
+                # Commit transaction or abort it
+                topic.producer.commit_transaction()
+                topic.index += 1
+                sent_count += 1
+                if topic.index % CONSUMER_LOGGING_THRESHOLD == 0:
+                    logger.debug(f"..sent {topic.index} to {topic.name}")
```

Comment:
I'm a bit confused here.. we are supposed to implement streams here? Something like..

https://github.com/redpanda-data/redpanda/pull/18790/files#diff-33bfdde92950a21be87d11b24ba3123f417c3dce8d236f5d1ab725eb66999b67

The idea is 

consume from one topic and produce atomically to other.. the idea of this verifier is we have a single topic with 'N' partitions each partition has contiguous integer sequence [x..x+n], when we atomically produce to an output topic with same number of partitions, the verifier should ensure the output topic also has the same sequence after atomically producing to the output topic (under various external conditions).

Reply:
Looks like I missed the third action here then: 'atomic_produce'. I'll implement it.